tox.ini

```diff
@@ -0,0 +1,67 @@
+[tox]
+envlist = pep8
+
+[testenv]
+deps =
+    -r{toxinidir}/requirements.txt
+    avocado-framework==94.0
+    python_magic
+setenv =
+    PYTHONPATH = {toxinidir}/tests
+
+[testenv:flake8]
+deps = flake8
+commands = flake8 sos tests
+
+[testenv:pylint]
+deps = pylint
+commands = pylint --rcfile=tox.ini sos tests
+
+[testenv:unit_tests]
+basepython = python3
+setenv =
+    PYTHONPATH = .
+commands =
+    avocado run tests/unittests/
+
+[testenv:stageone_tests]
+basepython = python3
+commands =
+    avocado run -p TESTLOCAL=true --test-runner=runner -t stageone tests/cleaner_tests
+    avocado run -p TESTLOCAL=true --test-runner=runner -t stageone tests/collect_tests
+    avocado run -p TESTLOCAL=true --test-runner=runner -t stageone tests/report_tests
+    avocado run -p TESTLOCAL=true --test-runner=runner -t stageone tests/vendor_tests
+
+[testenv:stagetwo_tests]
+basepython = python3
+commands =
+    avocado run -p TESTLOCAL=true --test-runner=runner -t stagetwo tests/cleaner_tests
+    avocado run -p TESTLOCAL=true --test-runner=runner -t stagetwo tests/collect_tests
+    avocado run -p TESTLOCAL=true --test-runner=runner -t stagetwo tests/report_tests
+    avocado run -p TESTLOCAL=true --test-runner=runner -t stagetwo tests/vendor_tests
+
+[testenv:foreman_tests]
+basepython = python3
+commands =
+    avocado run -p TESTLOCAL=true --test-runner=runner -t foreman tests/product_tests/foreman/
+
+[testenv:nosetests]
+basepython = python3
+deps =
+    {[testenv]deps}
+    nose3
+commands =
+    nosetests -v --with-coverage --cover-package=sos tests/unittests --cover-html
```

Comment:
We haven't used `nose` in a while, any particular reason behind bringing it back?

Reply:
Ah, I picked that one up from the tests that is run while the deb package is being built.

We don't have avocado in our repost yet, and this is some thing @dnegreira is going to correct in this cycle. Once that is done we can move to use avocado natively.

But this does work